libkernelflinger/log.c

```diff
@@ -169,5 +187,100 @@ void log(const CHAR16 *fmt, ...)
 
 	va_start(args, fmt);
 	vlog(fmt, args);
+
+	if (KERNELFLINGER_LOG_OUTPUT) {
+		log_file(fmt, args);
+	}
+
 	va_end(args);
 }
+
+#if DEBUG_MESSAGES
+EFI_FILE * log_init(const CHAR16 *filename)
+{
+
+    EFI_FILE *root;
+	EFI_FILE *file;
+    EFI_STATUS ret;
+
+	root = g_root;
+
+	if (!root)
+		return NULL;
+
+	file = NULL;
+
+	ret = uefi_call_wrapper(root->Open, 5, root, &file, (CHAR16 *)filename,
+				EFI_FILE_MODE_READ | EFI_FILE_MODE_WRITE | EFI_FILE_MODE_CREATE , 0);
+	if (EFI_ERROR(ret)) {
+		efi_perror(ret, L"Can't create logfile");
```

Comment:
Also print the filename:
efi_perror(ret, L"Can't create '%s' logfile", filename);

Reply:
Already modified.